### PR TITLE
[LW] Integration Tests

### PR DESCRIPTION
**Pull request by Jolyon-S**
**Goals (and why)**:
Fix equality checks, and add integration tests.

**Implementation Description (bullets)**:
* Integration tests
* Extract utility method for comparing `Map<Cell, byte[]`s
* fix cache value

**Testing (What was existing testing like?  What have you done to improve it?)**:
This PR is primarily about testing

**Concerns (what feedback would you like?)**:
Fixes the issues?

**Where should we start reviewing?**:
Diff

**Priority (whenever / two weeks / yesterday)**:
Yesterday, because of the fixes 🔥

### CODE REVIEW

File: timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+                .isExactlyInstanceOf(TransactionSerializableConflictException.class)
+                .hasMessageContaining("There was a read-write conflict on table");
+    }
+
+    @Test
+    public void serializableTransactionsDoNotThrowWhenOverwritingAPreviouslyCachedValue() {
+        putValue();
+
+        assertThatCode(() -> txnManager.runTaskThrowOnConflict(txn -> {
+                    txn.get(TABLE_REF, ImmutableSet.of(CELL_1));
```
Comment (gmaretic) [inline timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java:161]: I guess the first read wasn't actually cached here, we need another transaction first to get it into the cache

File: timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+                    txn.put(TABLE_REF, ImmutableMap.of(CELL_1, DATA_3));
+                    return null;
+                }))
+                .doesNotThrowAnyException();
+    }
+
+    @Test
+    public void serializableTransactionsReadViaTheCacheWherePossible() {
```
Comment (gmaretic) [inline timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java:169]: Maybe make more obvious that you mean the check reads from the cache

File: timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+        putValue();
+
+        txnManager.runTaskThrowOnConflict(txn -> {
```
Comment (gmaretic) [inline timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java:172]: Maybe add a test that we actually invalidate a cached entry once we write to it for the subsequent transactions?

**Review by gmaretic — COMMENTED** [review]
Tests seem pretty good, didn't check out the rest yet

File: timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java (hunk 1 above)
Comment (Jolyon-S) [reply timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java:161]: Yeah, I suppose so. It'd be locally cached after the get, but definitely worth actually making it be centrally cached.

**Pull request merged**